### PR TITLE
Url storage

### PR DESCRIPTION
**Pull request by ifedapoolarewaju**
This PR implements url storage functionality to the tusclient

### CODE REVIEW

File: tests/storage_file [hunk 1]
```diff
@@ -0,0 +1 @@
+{"_default": {"1": {"url": "http://master.tus.io/files/foo_bar", "key": "f3e5fa051d458fb7f22eaf03749c8272"}}}
```
Comment (Acconut) [inline tests/storage_file]: Is this file intended to be committed?

File: tusclient/uploader.py [hunk 2]
```diff
@@ -61,6 +63,17 @@ class Uploader(object):
         - retry_delay (int):
             How long (in seconds) the uploader should wait before retrying a failed upload attempt.
             If not specified, it defaults to 30.
+        - resumable (bool):
```
Comment (Acconut) [inline tusclient/uploader.py]: I am aware that the tus-js-client's option is named the same, but I would suggest renaming it to `storeURLs` or similar which is more expressive. I would love to do this for the tus-js-client but it's a breaking chance and is therefore not worth it.

File: tusclient/uploader.py [hunk 3]
```diff
@@ -61,6 +63,17 @@ class Uploader(object):
...
+            Determines whether or not url should be stored, and uploads should be resumed.
+        - storage (<tusclient.storage.interface.Storage>):
```
Comment (Acconut) [inline tusclient/uploader.py]: What about naming this `urlStorage`? Or else, this may be confused with the upload storage which is not correct.

File: tusclient/uploader.py [hunk 4]
```diff
@@ -61,6 +63,17 @@ class Uploader(object):
...
+            An implementation of <tusclient.storage.interface.Storage> which is an API for ur storage.
```
Comment (Acconut) [inline tusclient/uploader.py]: s/ur storage/URL storage/ :)

File: tusclient/uploader.py [hunk 5]
```diff
@@ -71,24 +84,34 @@ class Uploader(object):
...
+                 chunk_size=None, metadata=None, retries=0, retry_delay=30,
+                 resumable=False, storage=None, fingerprinter=None):
         if file_path is None and file_stream is None:
             raise ValueError("Either 'file_path' or 'file_stream' cannot be None.")
 
         if url is None and client is None:
             raise ValueError("Either 'url' or 'client' cannot be None.")
 
+        if resumable and storage is None:
+            raise ValueError("Please specify a storage instance to enable resumablility.")
```
Comment (Acconut) [inline tusclient/uploader.py]: The documentation above said that a default storage would be used here:
> if resumable is set to true, the default storage module (<tusclient.storage.filestorage.FileStorage>) would be used.

File: tusclient/fingerprint/fingerprint.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+            - fs[file]: The file stream instance of the file for which a fingerprint would be generated.
+        :Returns: fingerprint[str]
+        """
+        hasher = hashlib.md5()
+        # we encode the content to avoid python 3 uncicode errors
+        buf = self._encode_data(fs.read(self.BLOCK_SIZE))
+        while len(buf) > 0:
+            hasher.update(buf)
+            buf = fs.read(self.BLOCK_SIZE)
+        return hasher.hexdigest()
```
Comment (Acconut) [inline tusclient/fingerprint/fingerprint.py]: Would it make sense to return `"md5:" + haser.hexdigest()` to identify that this fingerprint is from a MD5?

File: tests/storage_file (hunk 1 above)
Comment (ifedapoolarewaju) [reply tests/storage_file]: yes it is. It's used to test the resumability feature. 🙂

File: tusclient/uploader.py (hunk 2 above)
Comment (ifedapoolarewaju) [reply tusclient/uploader.py]: agreed 👍

File: tusclient/uploader.py (hunk 5 above)
Comment (ifedapoolarewaju) [reply tusclient/uploader.py]: yes, that's wrong. I should remove it from the documentation.

**Comment by ifedapoolarewaju** [comment]
@Acconut I have made updates based on you feedback, please take a look 🙂 https://github.com/tus/tus-py-client/pull/12/commits/0dfe056e5db4248bf0625f2948d9aa57f23a14b5
